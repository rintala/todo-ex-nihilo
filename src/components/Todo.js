import React from "react";
import { useContext } from "react";
import "./../App.css";
import TodoContext from "./Provider";

const Todo = ({ todo, isLast }) => {
  const dispatch = useContext(TodoContext);

  const handleButtonCompleteClick = event => {
    dispatch({
      type: todo.complete ? "UNDO_TODO" : "DO_TODO",
      id: todo.id,
    });

    event.preventDefault();
  };

  const handleButtonRemoveClick = event => {
    dispatch({
      type: "REMOVE",
      id: todo.id,
    });

    event.preventDefault();
  };

  const handleCardClick = event => {
    dispatch({
      type: "SELECT_OR_UNSELECT",
      id: todo.id,
    });

    event.preventDefault();
  };

  return (
    <div
      key={todo.id}
      className={
        !isLast
          ? todo.selected
            ? "todoCardSelected"
            : "todoCard"
          : todo.selected
          ? "todoCardSelected todoCardSelectedLast"
          : "todoCard todoCardLast"
      }
      onClick={handleCardClick}
      onDoubleClick={handleButtonCompleteClick}
    >
      <div className="cardHeader">
        <li style={{ fontSize: "20px", fontWeight: "bold" }}>
          Task: {todo.task}
        </li>
        <button className="btnRemove" onClick={handleButtonRemoveClick}>
          X
        </button>
      </div>
      <ul>
        <li>ID: {todo.id}</li>
        {todo.dueDate !== null ? (
          <li>Deadline: {todo.dueDate.toString()}</li>
        ) : null}
        <li>Completed: {todo.complete.toString()}</li>
      </ul>
      <div className="cardFooter">
        {todo.complete ? (
          <button
            className="btnIsCompleted"
            onClick={handleButtonCompleteClick}
          >
            ✓
          </button>
        ) : (
          <button
            className="btnIsNotCompleted"
            onClick={handleButtonCompleteClick}
          >
            ✓
          </button>
        )}
      </div>
    </div>
  );
};

export default Todo;
