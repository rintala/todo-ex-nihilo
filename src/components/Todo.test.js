import React from "react";
import { create } from "react-test-renderer";
import Todo from "./Todo";

const todo = {
  id: "t0",
  task: "Complete tutorial on React hooks and state",
  complete: false,
  dueDate: new Date("July 9, 2019 00:00"),
  selected: false,
};

test("render correctly Todo Component", () => {
  const todoComponent = create(<Todo todo={todo} />).toJSON();
  expect(todoComponent).toMatchSnapshot();
});
