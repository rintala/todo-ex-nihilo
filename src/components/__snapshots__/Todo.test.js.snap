// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Todo Component 1`] = `
<div
  className="todoCard"
  onClick={[Function]}
  onDoubleClick={[Function]}
>
  <div
    className="cardHeader"
  >
    <li
      style={
        Object {
          "fontSize": "20px",
          "fontWeight": "bold",
        }
      }
    >
      Task: 
      Complete tutorial on React hooks and state
    </li>
    <button
      className="btnRemove"
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <ul>
    <li>
      ID: 
      t0
    </li>
    <li>
      Deadline: 
      Tue Jul 09 2019 00:00:00 GMT+0200 (Central European Summer Time)
    </li>
    <li>
      Completed: 
      false
    </li>
  </ul>
  <div
    className="cardFooter"
  >
    <button
      className="btnIsNotCompleted"
      onClick={[Function]}
    >
      âœ“
    </button>
  </div>
</div>
`;
